packages/opencensus-core/test/test-tracer.ts

```diff
@@ -0,0 +1,269 @@
+
+import {TracerConfig} from '../src/trace/config/types';
+import {RootSpan} from '../src/trace/model/root-span';
+import {Span} from '../src/trace/model/span';
+import {Tracer} from '../src/trace/model/tracer';
+import {OnEndSpanEventListener} from '../src/trace/model/types';
+
+class OnEndSpanClass implements OnEndSpanEventListener {
+  /** Counter for test use */
+  testCount = 0;
+  /** Happens when a span is ended */
+  onEndSpan(span: RootSpan): void {
+    this.testCount++;
+  }
+}
+
+const defaultConfig: TracerConfig = {
+  samplingRate: 1
+};
+
+describe('Tracer', () => {
+  const options = {name: 'test'};
+
+  /** Should create a Tracer instance */
+  describe('new Tracer()', () => {
+    it('should create a Tracer instance', () => {
+      const tracer = new Tracer();
+      assert.ok(tracer instanceof Tracer);
+    });
+  });
+
+  /** Should get/set the current RootSpan from tracer instance */
+  describe('get/set currentRootSpan()', () => {
+    let tracer, rootSpan;
+    before(() => {
+      tracer = new Tracer();
+      rootSpan = new RootSpan(tracer);
+      tracer.currentRootSpan = rootSpan;
+    });
+    it('should get the current RootSpan from tracer instance', () => {
+      assert.ok(tracer.currentRootSpan instanceof RootSpan);
+    });
+    it('should set the current RootSpan from tracer instance', () => {
+      assert.equal(tracer.currentRootSpan, rootSpan);
+    });
+  });
+
+  /** Should return a started tracer instance */
+  describe('start()', () => {
+    let tracerStarted;
+    before(() => {
+      const tracer = new Tracer();
+      tracerStarted = tracer.start(defaultConfig);
+    });
+    it('should return a tracer instance', () => {
+      assert.ok(tracerStarted instanceof Tracer);
+    });
+
+    it('the trace was started', () => {
+      assert.ok(tracerStarted.active);
+    });
+  });
+
+  /** Should return an OnEndSpanEventListener list */
+  describe('registerEndSpanListener() / get eventListeners()', () => {
+    let tracer, onEndSpan;
+    before(() => {
+      tracer = new Tracer();
+      onEndSpan = new OnEndSpanClass();
+      tracer.registerEndSpanListener(onEndSpan);
+    });
+
+    it('should register a new OnEndSpanEventListener on listners list', () => {
+      const listner = tracer.eventListeners[0];
+      assert.equal(listner, onEndSpan);
+    });
+
+    it('should return an OnEndSpanEventListener list', () => {
+      for (const listner of tracer.eventListeners) {
+        assert.ok(listner instanceof OnEndSpanClass);
+      }
+    });
+  });
+
+  /** Should stop the trace instance */
+  describe('stop()', () => {
+    it('should stop the trace instance', () => {
+      const tracer = new Tracer();
+      tracer.start(defaultConfig);
+      tracer.stop();
+      assert.ok(!tracer.active);
+    });
+  });
+
+  /** Should check if the trace instance is started or stop */
+  describe('get active()', () => {
+    let tracer: Tracer;
+    before(() => {
+      tracer = new Tracer();
```

Comment:
Add an assertion here to ensure that it wasn't `active` beforehand.

Reply:
Done !